newdle/client/src/components/creation/ParticipantsStep.js

```diff
@@ -1,26 +1,63 @@
 import React from 'react';
 import {useDispatch, useSelector} from 'react-redux';
-import {t} from '@lingui/macro';
+import {useHistory} from 'react-router';
+import {t, Trans} from '@lingui/macro';
+import PropTypes from 'prop-types';
 import {Button, Container, Icon} from 'semantic-ui-react';
-import {setStep} from '../../actions';
-import {areParticipantsDefined} from '../../selectors';
+import {setStep, updateNewdle} from '../../actions';
+import client from '../../client';
+import {areParticipantsDefined, getCreatedNewdle, getParticipants} from '../../selectors';
+import {STEPS} from './steps';
 import UserSearch from './userSearch';
 import styles from './creation.module.scss';
 
-export default function ParticipantsStep() {
+export default function ParticipantsStep({isEditing}) {
   const dispatch = useDispatch();
   const participantsDefined = useSelector(areParticipantsDefined);
+  const participants = useSelector(getParticipants);
+  const activeNewdle = useSelector(getCreatedNewdle);
+  const history = useHistory();
+  const [_editNewdle, submitting] = client.useBackendLazy(client.updateNewdle);
+
+  async function editNewdle() {
+    const newdle = await _editNewdle(activeNewdle.code, {participants});
+
+    if (newdle) {
+      dispatch(updateNewdle(newdle));
```

Comment:
We also need to `dispatch(clearParticipantCodes());` here - otherwise, there's an edge case where, if you delete yourself, and then click the "answer newdle" button in the header, you get your old participant code in the URL which no longer exists.

Reply:
Good point! Even while reinforcing it while clearing the participant code, we can also add some content to show the user's no longer a participant since this edge case is also applicable to users that might be have been invited before or with active codes.

Reply:
showing a nicer error message for invalid participant codes sounds good but please not in this PR :)